### PR TITLE
Feature/resolve query

### PR DESCRIPTION
**Pull request by ZhongFuze**
Multi-collection query for EdgeCollection ``Proofs`` and ``Holds``
1. neighbor(depth):
    add source ``dotbit``
2. neighbor_with_traversal(depth):
    abstract the same part of two collections

Resolves name domain:
```
query {
  ens(name: "xxxx.eth") {
    source
    fetcher
    system
    name
    resolved {
      category
      address
      chain
    }
    owner {
      identity
      platform
    }
  }
}
```

```
query {
  dotbit(name: "xxxx.bit") {
    system
    name
    resolved {
      category
      address
      chain
    }
    owner {
      identity
      platform
      neighbor(depth: 5) {...}
    }
  }
}

```

### CODE REVIEW

File: src/graph/edge/proof.rs [hunk 1]
```diff
@@ -136,6 +136,37 @@ impl From<DatabaseRecord<EdgeRecord<Proof>>> for ProofRecord {
...
+
+    pub uuid: Uuid,
+    pub source: DataSource,
+    pub created_at: Option<NaiveDateTime>,
+    pub updated_at: NaiveDateTime,
+    pub fetcher: DataFetcher,
+}
+
+impl IdentityFromToRecord {
+    pub fn id(&self) -> &String {
```
Comment (nykma) [inline src/graph/edge/proof.rs:165]: Return type should be `&str` instead of `&String`

**Review by nykma — CHANGES_REQUESTED** [review]
See comments

File: src/controller/graphql/resolve.rs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+use strum::IntoEnumIterator;
+use tracing::debug;
+use uuid::Uuid;
+
+// #[derive(SimpleObject)]
+// #[graphql(concrete(name = "IdentityRecord", params(IdentityRecord)))]
+// #[graphql(concrete(name = "ContractRecord", params(ContractRecord)))]
+
+#[async_trait::async_trait]
+pub trait ResolveE<RecordType> {
```
[the remark this answers is not in the document]
Comment (ZhongFuze) [reply src/controller/graphql/resolve.rs]: remove unused code

File: src/controller/graphql/resolve.rs [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    async fn fetcher(&self) -> DataFetcher {
+        self.fetcher
+    }
+
+    /// When this connection is fetched by us RelationService.
+    async fn updated_at(&self) -> i64 {
+        self.updated_at.timestamp()
+    }
+
+    async fn resolved(&self) -> Result<ContractRecord> {
```
[the remark this answers is not in the document]
Comment (ZhongFuze) [reply src/controller/graphql/resolve.rs:56]: done

**Review by nykma — APPROVED** [review]
:shipit:

**Pull request merged**